enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py

```diff
@@ -1719,13 +1723,85 @@ def test_credits_available_endpoint_with_content_assignments(
+        Test the can-request endpoint for subsidy access policy.
+        """
+        # Create a policy with BnR enabled
+        learner_credit_config = LearnerCreditRequestConfiguration.objects.create()
+        policy = AssignedLearnerCreditAccessPolicyFactory(
+            display_name="Test BnR Policy",
+            enterprise_customer_uuid=self.enterprise_uuid,
+            active=True,
+            spend_limit=1000000,
+            subsidy_uuid=uuid4(),
+            catalog_uuid=uuid4(),
+            learner_credit_request_config=learner_credit_config,
+        )
+        policy.learner_credit_request_config = learner_credit_config 
+        policy.save()
+        
+        # Set up the request
+        content_key = "course-v1:edX+Privacy101+3T2020"
+        lms_user_id = 1234
+        
+        url = reverse(
+            "api:v1:policy-redemption-can-request",
+            kwargs={"enterprise_customer_uuid": self.enterprise_uuid}
+        )
+        query_params = {
+            'content_key': content_key,
+            'lms_user_id': lms_user_id
+        }
+        # Make sure the learner_credit_config is active to enable BnR
+        learner_credit_config.active = True
+        learner_credit_config.save()
+        
+        # Mock the catalog_contains_content_key method to return True
+        with mock.patch(
+            'enterprise_access.apps.subsidy_access_policy.models.SubsidyAccessPolicy.catalog_contains_content_key',
+            return_value=True
+        ):
+            response = self.client.get(url, query_params)
+        self.assertEqual(response.status_code, 200)
+        response_data = response.json()
+        self.assertTrue(response_data.get('can_request'))
+        self.assertEqual(response_data.get('content_key'), content_key)
+        self.assertEqual(response_data.get('redeemable_subsidy_access_policy')['uuid'], str(policy.uuid))
+        
+        # Test when no BnR enabled policies exist
+        learner_credit_config.active = False
+        learner_credit_config.save()
+        
+        response = self.client.get(url, query_params)
+        self.assertEqual(response.status_code, 400)
+        response_data = response.json()
+        self.assertFalse(response_data.get('can_request'))
+        self.assertEqual(response_data.get('reason'), 'No policies with BnR enabled found')
+        
+        # Test when content is not in catalog
+        learner_credit_config.active = True
+        learner_credit_config.save()
+        
+        with mock.patch(
+            'enterprise_access.apps.subsidy_access_policy.models.SubsidyAccessPolicy.catalog_contains_content_key',
+            return_value=False
+        ):
+            response = self.client.get(url, query_params)
+        self.assertEqual(response.status_code, 400)
+        response_data = response.json()
+        self.assertFalse(response_data.get('can_request'))
+        self.assertEqual(response_data.get('reason'), REASON_CONTENT_NOT_IN_CATALOG)
```

Comment:
we should also cover the third usecase when learner request with the same course key already exists in any of the "NON RE-REQUESTABLE" states.

Reply:
Done!
Now we'll look for existing requests `state__in=[SubsidyRequestStates.REQUESTED, SubsidyRequestStates.APPROVED]`

Reply:
it should also include `ACCEPTED`